rewrite-java/src/main/java/org/openrewrite/java/cleanup/CombineSemanticallyEqualCatchBlocks.java

```diff
@@ -0,0 +1,323 @@
+/*
+ * Copyright 2022 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.cleanup;
+
+import org.openrewrite.*;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.java.JavaVisitor;
+import org.openrewrite.java.search.SemanticallyEqual;
+import org.openrewrite.java.tree.*;
+import org.openrewrite.marker.Markers;
+
+import java.time.Duration;
+import java.util.*;
+
+import static java.util.Collections.emptyMap;
+
+public class CombineSemanticallyEqualCatchBlocks extends Recipe {
+
+    @Override
+    public String getDisplayName() {
+        return "Combine semantically equal catch blocks";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Combine catches in a try that contain semantically equivalent blocks. " +
+                "No change will be made when a caught exception exists if combing catches may change application behavior or type attribution is missing.";
+    }
+
+    @Override
+    public Set<String> getTags() {
+        return Collections.singleton("RSPEC-2147");
+    }
+
+    @Override
+    public Duration getEstimatedEffortPerOccurrence() {
+        return Duration.ofMinutes(5);
+    }
+
+    @Override
+    protected TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new CombineSemanticallyEqualCatchBlocksVisitor();
+    }
+
+    private static class CombineSemanticallyEqualCatchBlocksVisitor extends JavaVisitor<ExecutionContext> {
+
+        // TODO: account for comments.
```

Comment:
@sambsnyd, you mentioned `SemanticallyEqual` should not contain a mechanism to consider comments while comparing equality or equivalency.

Without the changes, I need to copy `SemanticallyEqual` into `CombineSemanticallyEqualCatchBlocksVisitor` and make a few adjustments to compare comments between the J trees.

Do you think it's okay to copy the entire class into this recipe until we see a need to account for comments in `SemanticallyEqual`?